dll/win32/shdocvw/CFavBand.cpp

```diff
@@ -0,0 +1,648 @@
+    tbb[iButton].idCommand = ID_ORGANIZE;
+    tbb[iButton].fsState = TBSTATE_ENABLED;
+    tbb[iButton].fsStyle = BTNS_BUTTON | BTNS_AUTOSIZE | BTNS_SHOWTEXT;
+    tbb[iButton].iString = (INT)m_hwndToolbar.SendMessage(TB_ADDSTRING, 0, (LPARAM)szzOrganize);
+    ++iButton;
+    ATLASSERT(iButton == _countof(tbb));
+
+    LRESULT ret = m_hwndToolbar.SendMessage(TB_ADDBUTTONS, iButton, (LPARAM)&tbb);
+    ATLASSERT(ret);
+
+    return ret;
+}
+
+BOOL CFavBand::CreateTreeView()
+{
+    m_hTreeViewImageList = ImageList_Create(16, 16, ILC_COLOR32 | ILC_MASK, 64, 0);
+    ATLASSERT(m_hTreeViewImageList);
+    if (!m_hTreeViewImageList)
+        return FALSE;
+
+    DWORD style, exstyle;
+    style = TVS_NOHSCROLL | TVS_NONEVENHEIGHT | TVS_FULLROWSELECT | TVS_INFOTIP |
+            TVS_SINGLEEXPAND | TVS_TRACKSELECT | TVS_SHOWSELALWAYS | TVS_EDITLABELS |
+            WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN | WS_TABSTOP;
+    exstyle = WS_EX_CLIENTEDGE;
+
+    HWND hwndTV = ::CreateWindowExW(exstyle, WC_TREEVIEWW, NULL, style,
+                                    0, 0, 0, 0,
+                                    m_hWnd, (HMENU)(ULONG_PTR)IDW_TREEVIEW,
+                                    ::GetModuleHandleW(NULL), NULL);
+    ATLASSERT(hwndTV);
+    if (!hwndTV)
+        return FALSE;
+
+    m_hwndTreeView.Attach(hwndTV);
+    TreeView_SetImageList(m_hwndTreeView, m_hTreeViewImageList, TVSIL_NORMAL);
+
+    return TRUE;
+}
+
+// *** message handlers ***
+
+LRESULT CFavBand::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (!CreateToolbar() || !CreateTreeView())
+        return -1;
+
+    return 0;
+}
+
+LRESULT CFavBand::OnDestroy(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    m_hwndTreeView.DestroyWindow();
+    m_hwndToolbar.DestroyWindow();
+    return 0;
+}
+
+LRESULT CFavBand::OnSize(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (!m_hwndTreeView)
+        return 0;
+
+    RECT rc;
+    GetClientRect(&rc);
+    LONG cx = rc.right, cy = rc.bottom;
+
+    RECT rcTB;
+    m_hwndToolbar.SendMessage(TB_AUTOSIZE, 0, 0);
+    m_hwndToolbar.GetWindowRect(&rcTB);
+
+    LONG cyTB = rcTB.bottom - rcTB.top;
+    m_hwndTreeView.MoveWindow(0, cyTB, cx, cy - cyTB);
+
+    return 0;
+}
+
+LRESULT CFavBand::OnSetFocus(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    m_bFocused = TRUE;
+    IUnknown_OnFocusChangeIS(m_pSite, reinterpret_cast<IUnknown*>(this), TRUE);
+    return 0;
+}
+
+LRESULT CFavBand::OnKillFocus(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    IUnknown_OnFocusChangeIS(m_pSite, reinterpret_cast<IUnknown*>(this), FALSE);
+    m_bFocused = FALSE;
+    return 0;
+}
+
+LRESULT CFavBand::OnCommand(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    switch (LOWORD(wParam))
+    {
+        case ID_ADD:
+        {
+            UNIMPLEMENTED;
+            MessageBox(L"Not implemented yet", NULL, MB_ICONERROR);
```

Comment:
Missing translation. You can use the message box in shellutils.h and ERROR_NOT_IMPLEMENTED or another suitable error code.

Reply:
Fixed as `SHELL_ErrorBox(m_hWnd, ERROR_NOT_SUPPORTED);`.

Reply:
![error](https://github.com/user-attachments/assets/731d9705-91a0-40d4-a37c-7218ac014d0c)